### PR TITLE
Reg cocktails apt1.0+preprocessing

### PR DESCRIPTION
**Pull request by ravinkohli**

### CODE REVIEW

File: autoPyTorch/api/base_task.py [hunk 1]
```diff
@@ -122,6 +122,241 @@ def get_search_updates():
...
+        hyperparameter='SGDOptimizer:weight_decay',
+        value_range=[1e-5, 1e-1],
+        default_value=1e-3,
+    )
+    search_space_updates.append(
+        node_name='optimizer',
+        hyperparameter='SGDOptimizer:momentum',
+        value_range=[0.1, 0.999],
+        default_value=0.1,
+        log=True
```
Comment (ArlindKadra) [inline autoPyTorch/api/base_task.py:300]: Not log in the paper table

File: autoPyTorch/api/base_task.py [hunk 2]
```diff
@@ -122,6 +122,241 @@ def get_search_updates():
...
+    search_space_updates.append(
+        node_name='optimizer',
+        hyperparameter='SGDOptimizer:lr',
+        value_range=[1e-4, 1e-1],
+        default_value=1e-3,
+        log=True
+    )
+    search_space_updates.append(
+        node_name='optimizer',
+        hyperparameter='SGDOptimizer:weight_decay',
```
Comment (ArlindKadra) [inline autoPyTorch/api/base_task.py:291]: weight decay for adam?

File: autoPyTorch/api/base_task.py [hunk 3]
```diff
@@ -122,6 +122,241 @@ def get_search_updates():
...
+        value_range=[16, 512],
+        default_value=128,
+        log=True
+    )
+
+    # preprocessing
+    search_space_updates.append(
+        node_name='feature_preprocessor',
+        hyperparameter='TruncatedSVD:target_dim',
+        value_range=[0.1, 0.9],
```
Comment (ArlindKadra) [inline autoPyTorch/api/base_task.py:314]: Did this change to percentage-wise from the old implementation ?

File: autoPyTorch/api/base_task.py (hunk 2 above)
Comment (ArlindKadra) [reply autoPyTorch/api/base_task.py:291]: Also weight decay should be turned on as `use_weight_decay` to `True`

**Review by ArlindKadra — COMMENTED** [review]
2 notes except my other comments. I will take it as we have the last layer `output_dimension`, we have 1 less layer/group. Additionally, `max_dropout` should be changed to [0,1]

**Comment by ArlindKadra** [comment]
Additionally the dropout shape should be `funnel`

File: autoPyTorch/api/base_task.py [hunk 4]
```diff
@@ -122,6 +122,241 @@ def get_search_updates():
...
+    # mlp backbone
+    search_space_updates.append(
+        node_name='network_backbone',
+        hyperparameter='ShapedMLPBackbone:mlp_shape',
+        value_range=['funnel'],
+        default_value='funnel',
+    )
+    search_space_updates.append(
+        node_name='network_backbone',
+        hyperparameter='ShapedMLPBackbone:num_groups',
```
[the remark this answers is not in the document]
Comment (ravinkohli) [reply autoPyTorch/api/base_task.py:227]: yeah it should be layers but its groups right now

File: autoPyTorch/api/base_task.py (hunk 1 above)
Comment (ravinkohli) [reply autoPyTorch/api/base_task.py:300]: oh thanks.

File: autoPyTorch/api/base_task.py (hunk 3 above)
Comment (ravinkohli) [reply autoPyTorch/api/base_task.py:314]: yes

File: autoPyTorch/api/base_task.py (hunk 2 above)
Comment (ravinkohli) [reply autoPyTorch/api/base_task.py:291]: yeah I have added it for adam as well

**Comment by ravinkohli** [comment]
> additionally if I am not mistaken for Lucas's code the beta1 and beta2 were also fixed for adam.

True they did not have them in the search space. They were using pytorch defaults

**Review by ArlindKadra — APPROVED** [review]
